src/extra/gpkg.jl

```diff
@@ -0,0 +1,1088 @@
+Tables.getcolumn(gt::GPKGTable, nm::Symbol) = Tables.getcolumn(gt.table, nm)
+Tables.schema(gt::GPKGTable) = Tables.schema(gt.table)
+
+# Implement GeoInterface.crs method
+GI.crs(gt::GPKGTable) = gt.crs
+
+const WKB_POINT = 0x00000001
+const WKB_LINESTRING = 0x00000002
+const WKB_POLYGON = 0x00000003
+const WKB_MULTIPOINT = 0x00000004
+const WKB_MULTILINESTRING = 0x00000005
+const WKB_MULTIPOLYGON = 0x00000006
+const WKB_GEOMETRYCOLLECTION = 0x00000007
+
+const WKB_Z = 0x80000000
+const WKB_M = 0x40000000
+const WKB_ZM = 0xC0000000
+
+function parse_gpb_header(blob::Vector{UInt8})
+  io = IOBuffer(blob)
+  
+  magic = read(io, UInt16)
+  magic == 0x5047 || error("Invalid GeoPackage binary header")
+  
+  version = read(io, UInt8)
+  flags = read(io, UInt8)
+  
+  srs_id = ltoh(read(io, Int32))
+  
+  envelope_type = flags >> 1 & 0x07
+  has_envelope = envelope_type > 0
+  
+  min_z = max_z = min_m = max_m = nothing
+  
+  if has_envelope && envelope_type >= 1
+    min_x = ltoh(read(io, Float64))
+    max_x = ltoh(read(io, Float64))
+    min_y = ltoh(read(io, Float64))
+    max_y = ltoh(read(io, Float64))
+  else
+    # Default envelope when not present
+    min_x = max_x = min_y = max_y = 0.0
+  end
+  
+  if has_envelope
+    if envelope_type >= 2
+      min_z = ltoh(read(io, Float64))
+      max_z = ltoh(read(io, Float64))
+    end
+    if envelope_type >= 3
+      min_m = ltoh(read(io, Float64))
+      max_m = ltoh(read(io, Float64))
+    end
+  end
+  
+  header = GeoPackageBinaryHeader(
+    magic, version, flags, srs_id,
+    min_x, max_x, min_y, max_y,
+    min_z, max_z, min_m, max_m
+  )
+  
+  return header, position(io)
+end
+
+function read_coordinate_values(io::IOBuffer, byte_order::UInt8, has_z::Bool, has_m::Bool)
+  read_value = byte_order == 0x01 ? ltoh : ntoh
+  
+  x = read_value(read(io, Float64))
+  y = read_value(read(io, Float64))
+  
+  coords_list = [x, y]
+  
+  if has_z
+    z = read_value(read(io, Float64))
+    push!(coords_list, z)
+  end
+  
+  if has_m
+    m = read_value(read(io, Float64))
+  end
+  
+  return coords_list
+end
+
+function create_point_from_coords(coords_list::Vector{Float64}, crs_type)
+  x, y = coords_list[1], coords_list[2]
+  
+  if isnan(x) && isnan(y)
+    return nothing
+  end
+  
+  # Handle different CRS types properly
+  if isnothing(crs_type)
+    # No CRS specified - use raw coordinates as Cartesian
+    return Point(coords_list...)
+  end
+  
+  # Determine CRS characteristics using CoordRefSystems.jl
+  try
+    # For geographic coordinate systems (LatLon family)
+    if crs_type <: LatLon
+      # WKB stores geographic coordinates as x=longitude, y=latitude
+      # LatLon constructor expects (latitude, longitude) order
+      if length(coords_list) == 2
+        return Point(crs_type(y, x))  # LatLon(lat, lon) from WKB(x=lon, y=lat)
+      else
+        # 3D geographic with altitude/elevation
+        return Point(crs_type(y, x, coords_list[3]))
+      end
```

Comment:
This logic is flawed. The LatLon type only has 2 coordinates. If you are in this branch of the code, it doesn't make sense to check for a different number of coordinates. In other words, this branch should never be reached.

The logic should be purely based on CRS types.